### ISSUE TITLE
WebChat Blazor Component

### ISSUE THREAD

**Issue by Janisku7** [microsoft/BotFramework-WebChat#2125]
## Feature Request

**Is your feature request related to a problem? Please describe.**

Doesn't have webassembly options for WebChat

**Describe the suggestion or request in detail**

Now that Blazor is here and Webassembly getting more and more users. as BotFramework build using .NetCore too so Blazor Component could be strongly typed and more design for bot 

**Describe alternatives you have considered**

one thing to make WebChat blazor component myself

**Additional context**

[Blazor](https://blazor.net)

[Enhancement]

**Comment by corinagum** [comment]
My impression is that this might be considered a sample request. Does the rest of the team have thoughts?
@cwhitten, @compulim, @EricDahlvang :)

**Comment by EricDahlvang** [comment]
I don't know enough about WebAssembly to speak intelligently.  But this does sound like a great idea!

**Comment by compulim** [comment]
ASP.NET Razor/Blazor is very heavy on server, but light on client. This is traditional web development where every client commands are processed on the server (or WebAssembly in Blazor case). I.e. very few JavaScript.

In Web Chat, we opt for a different approach: heavy on client and light on server. This is commonly known as a modern approach for web development.

I don't see lots of values in building Web Chat in Razor/Blazor because:

- UI layout and performance is the strength of Web Chat. And it won't be easy to recreate this experience without the careful design of DOM tree and intensive use of JavaScript
- There are more developers interested in React than ASP.NET. It is easier for everyone to jump in and help shaping the future of Web Chat
- I believe Blazor is a stopgap tool to enable Razor in mobile-offline scenario, not a framework designed to build a modern web app from ground up (yet)

When people look at Web Chat, they think it is "rendering bubbles." But there are lots more about it:

- Accessibility, we target WCAG 2.1 with a careful design on DOM tree
- Speech, we use W3C Web Speech API and support any engines that use the standard
- Total UI customization, you can throw away our UI and rebuild just using our business logic, without forking/hacking our code
   - You can build Web Chat as a CLI tool using Node.js
   - You can build a real-time shape-shifting experience, i.e. from full transcript (today), to single message (turn-based), to speech only and no visual UI (voice assistant)
   - You can build an offline experience
- Wide platform support, from IE11 to desktop/mobile browsers with touch/stylus to Xbox (i.e. scroll using a joystick)

In conclusion, I think it is not effective to re/build Web Chat using Blazor given its current direction.

**Comment by EricDahlvang** [comment]
Thank you for the breakdown @compulim !

**Comment by Alienroid** [comment]
blazor wasm is heavy on client and light on server (if there's one). You are referring to Blazor Server Side.

**Comment by compulim** [comment]
@Alienroid you are right. Blazor is WASM-ed .NET with ASP.NET server built-in, very heavy on client.
